orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -484,10 +500,11 @@ Status OrtLoadOptimizerStatesInternal(const PathString& optimizer_folder_path,
 Status OrtLoadCustomPropertyInternal(const PathString& property_folder_path,
                                      PropertyBag& property_bag) {
   // Find custom property files.
-  std::vector<std::string> custom_property_filenames;
+  std::vector<PathString> custom_property_filenames;
   FilterFilesFromDirectory(
       property_folder_path,
-      [&custom_property_filenames](const std::string& filename_str) -> bool {
+      [&custom_property_filenames](const PathChar* filename) -> bool {
+        PathString filename_str = filename;
```

Comment:
can we not take in const filename so that we can avoid this copy

Reply:
This is the lamda type defined by LoopDir function, I am afraid we have to modify its usage everywhere if we want to do that. I would prefer not change that, at least in our poc branch.